deployment/terraform/resources/apim.tf

```diff
@@ -168,6 +192,40 @@ resource "azurerm_api_management_api_policy" "pctasks_policy" {
         <set-header name="host" exists-action="override">
             <value>@(context.Request.OriginalUrl.ToUri().Host)</value>
         </set-header>
+        <!--
+            If an optional Authorization header is provided, validate it and set the X-Has-Authorization token. Otherwise,
+            ensure the header is overridden to "false". The backend code will need to validate that a subscription key or
+            auth token was provided via the headers in this policy.
+        -->
+        <choose>
+            <when condition="@(context.Request.Headers.GetValueOrDefault("Authorization", "") != "")">
+                <validate-jwt header-name="Authorization" failed-validation-error-message="Provided authorization token is invalid" require-expiration-time="true" require-scheme="Bearer" require-signed-tokens="true" output-token-variable-name="jwt">
+                    <!--JWT must be signed using keys from originating app tenant -->
+                    <openid-config url="https://login.microsoftonline.com/{{pctasks-apim-jwt-signing-tentant-id}}/v2.0/.well-known/openid-configuration" />
+                    <issuers>
+                        <issuer>https://login.microsoftonline.com/{{pctasks-apim-jwt-signing-tentant-id}}/v2.0</issuer>
+                    </issuers>
+                    <required-claims>
+                        <!--JWT must provide correct audience claim from the API backend app registration -->
+                        <claim name="aud">
+                            <value>{{pctasks-apim-jwt-aud-backend-app-id}}</value>
+                        </claim>
+                        <!--JWT must provide Run Read and Write API scopes as published by the app registration -->
+                        <claim name="scp">
+                            <value>Runs.Read.All Runs.Write.All</value>
```

Comment:
This may need to be rescoped to just Read.All

Reply:
This was correct in the end. The backend app reg is configured to pre-authorize both scopes to the frontend, even though the frontend didn't request the scope. I've updated the policy to better parse multiple scope entries out of a space delimited field, however both scopes are still included.